### PR TITLE
Added migrations

### PR DESCRIPTION
**Pull request by NJABULO-NDLOVU**
## Description

Added alembic migrations to the existing database.

## Types of changes

_Put an `x` in the boxes that apply._

- [ ] Bugfix (`fix`): A non-breaking change which fixes an issue or bug
- [x] New Feature (`feat`): A non-breaking change which adds a new feature or new functionality
- [ ] Tests (`test`): Adding missing tests or correcting existing tests
- [ ] Refactoring (`refactor`): A code change that neither fixes a bug nor adds a feature, e.g Deleting unused or redundant code
- [ ] Documentation (`docs`): Documentation only changes
- [ ] Other (`style`, `build`, `ci`, `performance`): Formatting, linting or styling. Changes to the CI pipeline. Performance improvements. Changes that affect the build system.

## Further Information

JIRA Ticket: <https://uxi.atlassian.net/browse/BEP-XXX>

Any relevant logs, screenshots, testing output, etc...

### CODE REVIEW

File: celery_sqlalchemy_scheduler/db/migrations/script.py.mako [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"""${message}
```
Comment (jengelhard) [inline celery_sqlalchemy_scheduler/db/migrations/script.py.mako:1]: I've never seen a .mako file before, but a quick google tells me Mako is a template library. When and why do we use it and why is it used here?

File: celery_sqlalchemy_scheduler/db/migrations/versions/2022_03_08_added_interval_schedule_table.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+"""added interval schedule table
```
Comment (CaptainCuddleCube) [inline celery_sqlalchemy_scheduler/db/migrations/versions/2022_03_08_added_interval_schedule_table.py:1]: why have multiple migrations in the same PR? Why not just roll them up?

File: celery_sqlalchemy_scheduler/db/migrations/env.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    )
+
+    with connectable.connect() as connection:
+        context.configure(
+            connection=connection,
+            target_metadata=target_metadata,
+            version_table_schema="scheduler",
+        )
+
+        # First ensure we have our scheduler schema available as all of these migrations will live
```
Comment (CaptainCuddleCube) [inline celery_sqlalchemy_scheduler/db/migrations/env.py:67]: In terms of storing this in a different schema, it can be a little confusing to people.. 

Why do you want the schedules to live in a different schema?

File: celery_sqlalchemy_scheduler/db/migrations/env.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+config = context.config
+
+# Interpret the config file for Python logging.
+# This line sets up loggers basically.
+fileConfig(config.config_file_name)
+
+# add your model's MetaData object here
+# for 'autogenerate' support
+# from myapp import mymodel
+# target_metadata = mymodel.Base.metadata
```
Comment (CaptainCuddleCube) [inline celery_sqlalchemy_scheduler/db/migrations/env.py:19]: Do we need these comments?

File: alembic.ini [hunk 5]
```diff
@@ -0,0 +1,105 @@
+# A generic, single database configuration.
```
Comment (CaptainCuddleCube) [inline alembic.ini:1]: 🤨 ; is this still generic?

File: celery_sqlalchemy_scheduler/db/migrations/script.py.mako (hunk 1 above)
Comment (NJABULO-NDLOVU) [reply celery_sqlalchemy_scheduler/db/migrations/script.py.mako:1]: template for what is suppose to be in a migrations file

File: celery_sqlalchemy_scheduler/db/migrations/versions/2022_03_08_added_interval_schedule_table.py (hunk 2 above)
Comment (NJABULO-NDLOVU) [reply celery_sqlalchemy_scheduler/db/migrations/versions/2022_03_08_added_interval_schedule_table.py:1]: that looked scary and bad

File: celery_sqlalchemy_scheduler/db/migrations/env.py (hunk 3 above)
Comment (NJABULO-NDLOVU) [reply celery_sqlalchemy_scheduler/db/migrations/env.py:67]: so that the tables and migrations aren't service dependant

File: celery_sqlalchemy_scheduler/db/migrations/env.py (hunk 4 above)
Comment (NJABULO-NDLOVU) [reply celery_sqlalchemy_scheduler/db/migrations/env.py:19]: don't know we always leave them in

Comment (NJABULO-NDLOVU) [reply celery_sqlalchemy_scheduler/db/migrations/env.py:19]: don't know we always leave them in

File: alembic.ini (hunk 5 above)
Comment (NJABULO-NDLOVU) [reply alembic.ini:1]: yup

File: celery_sqlalchemy_scheduler/db/migrations/versions/2022_03_08_added_interval_schedule_table.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "celery_interval_schedule",
+        sa.Column("id", sa.Integer(), autoincrement=True, nullable=False),
+        sa.Column("every", sa.Integer(), nullable=False),
```
Comment (stephen-aruba) [inline celery_sqlalchemy_scheduler/db/migrations/versions/2022_03_08_added_interval_schedule_table.py:23]: include some indication of unit, possibly in the column's 'comment' field if not the column name (or is this the unit...?)

File: celery_sqlalchemy_scheduler/db/migrations/versions/2022_03_08_added_periodic_task_tables.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "celery_periodic_task",
+        sa.Column("id", sa.Integer(), autoincrement=True, nullable=False),
+        sa.Column("name", sa.String(length=255), nullable=True),
+        sa.Column("task", sa.String(length=255), nullable=True),
+        sa.Column("interval_id", sa.Integer(), nullable=True),
+        sa.Column("crontab_id", sa.Integer(), nullable=True),
+        sa.Column("solar_id", sa.Integer(), nullable=True),
```
Comment (stephen-aruba) [inline celery_sqlalchemy_scheduler/db/migrations/versions/2022_03_08_added_periodic_task_tables.py:27]: I'm intrigued by the solar aspect... Is this related to sunrise or solar noon? We should chat 😊 ☀️

File: celery_sqlalchemy_scheduler/db/models.py [hunk 8]
```diff
@@ -1,20 +1,22 @@
...
+from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import Session, class_mapper, foreign, relationship, remote
 from sqlalchemy.sql import insert, select, update
 
-from .session import ModelBase
-from .tzcrontab import TzAwareCrontab
+from celery_sqlalchemy_scheduler.tzcrontab import TzAwareCrontab
 
-logger = get_logger("celery_sqlalchemy_scheduler.models")
+logger = get_logger("celery_sqlalchemy_scheduler.db.models")
```
Comment (stephen-aruba) [inline celery_sqlalchemy_scheduler/db/models.py:17]: Provides the module name automatically

```suggestion
logger = logging.getLogger(__name__)
```

**Review by stephen-aruba — COMMENTED** [review]
Couple comments.
One general comment, does this repo use Pipfiles (pipenv) or requirements (pip-tools) for dependencies?

**Review by stephen-aruba — APPROVED** [review]
LGTM 👍
